addons/web_editor/static/src/js/editor/image_processing.js

```diff
@@ -263,12 +263,29 @@ async function applyModifications(img, dataOptions = {}) {
  * @param {String} src URL of the image to load
  * @param {HTMLImageElement} [img] img element in which to load the image
  * @returns {Promise<HTMLImageElement>} Promise that resolves to the loaded img
+ *     or a placeholder image if the src is not found.
  */
 function loadImage(src, img = new Image()) {
-    return new Promise((resolve, reject) => {
-        img.addEventListener('load', () => resolve(img), {once: true});
-        img.addEventListener('error', reject, {once: true});
+    const handleImage = (src, resolve, reject) => {
+        img.addEventListener("load", () => resolve(img), {once: true});
+        img.addEventListener("error", reject, {once: true});
         img.src = src;
+    };
+    // The server will return a placeholder image with the following src.
+    const placeholderHref = "/web/image/__odoo__unknown__src__/";
```

Comment:
Whenever we end up using this placeholder, shouldn't we add a toaster to the client to let him know that he/someone/something deleted his original images and that the options won't work as expected?

Eg, I uploaded a tiger image, then added a nice shape and a nice filter:
![image](https://github.com/odoo/odoo/assets/30048408/f500a9d8-72ee-4c94-964b-1b8953e227da)

I deleted the original img
I then click on the image and see 
![image](https://github.com/odoo/odoo/assets/30048408/1fb4ebf1-c952-41b6-948e-82c449b705bc)

If I click on the cross icon next to the shape option to remove it, I end up with
![image](https://github.com/odoo/odoo/assets/30048408/c9db8b74-434c-45c4-b536-b78870d10579)

Which is more than fine considering he deleted his original image, but we could at least let him know (when he click on the image I suppose, which is when loadImage is called?): "Hey, it looks like the original image was deleted, using options might not behaves as expected. Please reupload the image if needed" something like that

Reply:
Yes it would be good to inform the user. Unfortunately, to display a notification, I need a `Widget` and here I don't have one. As we are in stable, I can't change the signature of the function to include a widget.. :cry: I propose to do this change in the master fw port :+1:

Reply:
Go with an alert then imo :grimacing: 
Not ideal but better than not being aware of what's going on and thinking the software if bugging no?